ArduPlane/altitude.cpp

```diff
@@ -361,26 +361,39 @@ int32_t Plane::calc_altitude_error_cm(void)
 }
 
 /*
-  check for FBWB_min_altitude_cm violation
- */
-void Plane::check_fbwb_minimum_altitude(void)
+  check for fence max altitude 
+*/
+void Plane::check_fbwb_altitude(void)
 {
-    if (g.FBWB_min_altitude_cm == 0) {
+    // fence max and min altitude (margin taken)
+    float max_alt_cm = fence.get_safe_alt_max()*100.0f;
+    float min_alt_cm = fence.get_safe_alt_min()*100.0f;
+    bool should_check_max = ((fence.get_enabled_fences() & AC_FENCE_TYPE_ALT_MAX) != 0) && (max_alt_cm > 0);
+    bool should_check_min = (((fence.get_enabled_fences() & AC_FENCE_TYPE_ALT_MIN) != 0) && (min_alt_cm > 0)) || (g.FBWB_min_altitude_cm != 0);
```

Comment:
it looks like it we have a FBWB min alt set, then we will use the fence min even if fence min is disabled

Reply:
@tridge ,Thanks for pointing that out.